app/main.py

```diff
@@ -1,25 +1,81 @@
-from abc import ABC
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: int | str) -> None:
```

Comment:
The `name` parameter in the `__set_name__` method should be of type `str`, not `int | str`. This method is called with the attribute name as a string.